Expensify/App
# Attachment Downloader

@jboniface, @laurenreidexpensify  please review this PR for issue [1040](https://github.com/Expensify/Expensify.cash/issues/1040)

### Details
<Explanation of the change or anything fishy that is going on>
The attachment modal in chat now provides an icon that can be clicked to download the image opened in modal.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes /issues/1040

### Tests

The attachment download feature was tested on below mentioned platforms and following steps can be followed to perform the test on any platform. This feature could not be tested on physical iOS device because its testing require provisioning profile that should include uuid of the testing device. So, any contributor who has that profile can help testing it.
1. Log into Expensify Cash
2. Click the paperclip icon
3. Select an image and confirm the upload
4. Once the image uploads to the chat, click the image to open in modal
5. Click on the download icon at the top right corner of the modal
6. Verify that image gets downloaded 
    a. For mobile apps check downloads/documents folder of device
    b. For desktop app the user will be prompted to save the file in desired folder
    c. For web app check the download manager of the browser

### Tested On


### Screenshots
<Add any necessary screenshots for all platforms if your change added or updated UI>

![attachment-downloader-web-chrome](https://user-images.githubusercontent.com/43338645/104037327-66986480-51f6-11eb-9051-eb7a84c85c8c.png)

#### Web

(Chrome)
![attachment-downloader-web-chrome](https://user-images.githubusercontent.com/43338645/104037327-66986480-51f6-11eb-9051-eb7a84c85c8c.png)

#### Mobile Web


#### Desktop
(MacOS)
![attachment-downloader-desktop-mac](https://user-images.githubusercontent.com/43338645/104037369-76b04400-51f6-11eb-98b3-4fede1ef9a93.png)

#### iOS

(Simulator)
![attachment-downloader-iOS-simulator](https://user-images.githubusercontent.com/43338645/104037496-8fb8f500-51f6-11eb-9fd3-0b86d204ff0b.png)

#### Android

(Physical Device)
![attachment-downloader-android-device](https://user-images.githubusercontent.com/43338645/104037409-862f8d00-51f6-11eb-80c6-a1a352f45505.jpeg)

**I have read the CLA Document and I hereby sign the CLA**
